src/plugins/formulas/formulas.js

```diff
@@ -203,36 +250,231 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    const hfSetUpProperly = this.setupHF();
+
+    if (!hfSetUpProperly) {
+      return;
+    }
+
+    // HyperFormula events:
+    this.hyperformula.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
+
+    this.applyHFSettings();
+
     super.enablePlugin();
   }
 
   /**
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
-    // TODO add tests for this line
-    this.hyperformula.destroy();
-
     super.disablePlugin();
   }
 
+  /**
+   * Triggered on `updateSettings`.
+   */
+  updatePlugin() {
+    this.#settings = this.hot.getSettings()[PLUGIN_KEY];
+
+    this.applyHFSettings();
+
+    if (isDefined(this.#settings.sheetName) && this.#settings.sheetName !== this.sheetName) {
+      this.switchSheet(this.#settings.sheetName);
+    }
+
+    super.updatePlugin();
+  }
+
   /**
    * Destroys the plugin instance.
    */
   destroy() {
+    if (this.hyperformula) {
```

Comment:
Yeah, if we removed the global register and relied on the garbage collector, this logic would not be needed anymore. I _guess_ we could do that? I'm always extra-careful with GC though, as it's real easy to leave some references behind and cause memory issues.

Reply:
On second thought, I think we should keep the `destroy` logic anyway. All things considered, Hyperformula is still a (semi)external library, and we won't know for sure if GC can handle everything they created.
Also, in my latest updates I changed most of the mentions of `hyperformula` in the code to `engine`, so it would be easier to change/add engines to the plugin. (for example, some legacy engine without HF even?). Keeping in mind, that `this.engine` could be potentially anything (wrapped in some API-translating logic compatible to HF), calling `destroy` on it is a good precaution, I think.

Reply:
We probably won't really need it, but sure, we can keep this logic.